dhis-2/dhis-api/src/main/java/org/hisp/dhis/programstageworkinglistdefinition/ProgramStageWorkingListDefinition.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2004-2022, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.programstageworkinglistdefinition;
+
+import java.util.Objects;
+
+import lombok.Setter;
+
+import org.hisp.dhis.common.BaseIdentifiableObject;
+import org.hisp.dhis.common.DxfNamespaces;
+import org.hisp.dhis.common.MetadataObject;
+import org.hisp.dhis.program.Program;
+import org.hisp.dhis.program.ProgramStage;
+import org.hisp.dhis.translation.Translatable;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlRootElement;
+
+@Setter
+@JacksonXmlRootElement( localName = "programStageWorkingListDefinition", namespace = DxfNamespaces.DXF_2_0 )
+public class ProgramStageWorkingListDefinition extends BaseIdentifiableObject
```

Comment:
I see you use the name in different ways different places: ProgramStageWorkingList and ProgramStageWorklingListDefinition.

I would prefer the first everywhere, making it a bit shorter and easier to read.

Is there any specific reason to include "Definition"?

Reply:
The reason to include "definition" at the end is to make clear that this is not a working list, this is just its "metadata", it's just a bunch of properties that will be applied to filter and sort a working list accordingly.

Reply:
I also would be in favor to remove the "definition" part from the name.
All the metadata that we have do not use "definition" to make it clear that is a metadata, having it just here for me increase the confusion instead of make things more clear.
On the other end, if you want to differentiate between data and metadata, we often use the "Instance" in the data to make it clear that is an instance of such metadata

Reply:
@muilpp: @enricocolasante makes an excellent point. Skipping "Definition" makes sense to me now.

Reply:
I've removed the word definition, now it's called programStageWorkingList everywhere